virttest/remote_commander/remote_runner.py

```diff
@@ -295,6 +297,10 @@ def parse_func_name(self, func_name, commander):
         :param func_name: Function name
         :param commander: Where to execute the command (remote or local)
         """
+        for name in func_name:
```

Comment:
Well this actually does nothing. I just tried it with using `commander.async`. This sets `self.manage = True`, but on line `307` it also sets `self.async`, which makes the command fail.

So I suggest raising an exception when trying to encapsulate it:

```diff
diff --git a/virttest/remote_commander/remote_master.py b/virttest/remote_commander/remote_master.py
index e0ed286..bf8d865 100644
--- a/virttest/remote_commander/remote_master.py
+++ b/virttest/remote_commander/remote_master.py
@@ -148,6 +148,11 @@ class CmdEncapsulation(object):
     """
 
     def __init__(self, master, obj_name, name):
+        if name is "python_file_run_with_helper" and obj_name != ["manage"]:
+            raise NotImplementedError("The 'python_file_run_with_helper' is "
+                                      "only implemented for 'manage' target. "
+                                      "Please use 'commander.manage.python_fi"
+                                      "le_run_with_helper' to run this.")
         self.master = master
         if obj_name is None:
             self.name = [name]
diff --git a/virttest/remote_commander/remote_runner.py b/virttest/remote_commander/remote_runner.py
index efa042b..0d246c7 100755
--- a/virttest/remote_commander/remote_runner.py
+++ b/virttest/remote_commander/remote_runner.py
@@ -297,10 +297,6 @@ class CmdSlave(object):
         :param func_name: Function name
         :param commander: Where to execute the command (remote or local)
         """
-        for name in func_name:
-            if name == "python_file_run_with_helper":
-                # Always run external test in the current process.
-                self.manage = True
         if func_name[0] == "manage":  # start command in main process.
             self.manage = True
             func_name = func_name[1:]
```

With this I have nothing against. Still you have to gain yet another ack to get it merged...